File: LeetCode/Python Solutions/Reverse Words in a String III/reverse.py
class Solution:
    def reverseWords(self, s: str) -> str:
        words = s.split(' ')

        return ' '.join(word[::-1] for word in words)


if __name__ == '__main__':
    Instant = Solution()
    Solve = Instant.reverseWords(s = "Let's take LeetCode contest")
    print(Solve)
    # s = "Let's take LeetCode contest" -> "s'teL ekat edoCteeL tsetnoc"

File: LeetCode/Java Solutions/Reverse Words in a String III/reverse.java
public class Solution {
    public String reverseWords(String s) {
        String[] words = s.split(" ");
        
        for (int i = 0; i < words.length; i++) {
            words[i] = new StringBuilder(words[i]).reverse().toString();
        }
        
        return String.join(" ", words);
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        
        String s1 = "Let's take LeetCode contest";
        String s2 = "God Ding";
        
        System.out.println(solution.reverseWords(s1)); // Expected output: "s'teL ekat edoCteeL tsetnoc"
        System.out.println(solution.reverseWords(s2)); // Expected output: "doG gniD"
    }
}
